File: Data Structures & Algorithms/palindrome-partitioning/submission-0.cpp
class Solution {
public:
vector<vector<string>> res;
bool isPalindrome(string s){
       int n=s.size();
       for(int i=0;i<n/2;i++){
        if(s[i]!=s[n-i-1]) return false;
       }
       return true;
}
void func(string s,int i,vector<string>& cur,string& temp){
    if(i>=s.size()){
        res.push_back(cur);
        return;
    }
    for(int j=i;j<s.size();j++){
        string temp=s.substr(i,j-i+1);
        if(isPalindrome(temp)){
            cur.push_back(temp);
            func(s,j+1,cur,temp);
            cur.pop_back();
        }
    }

}
    vector<vector<string>> partition(string s) {
        vector<string> cur;
        string temp;
        func(s,0,cur,temp);
        return res;
    }
};

File: Data Structures & Algorithms/valid-palindrome-ii/submission-0.py
class Solution:
    def isPalindrome(self,s:str) -> bool :
        l=0
        r=len(s)-1
        while l < r :
            if s[l] != s[r] : return False
            else :
                l+=1
                r-=1
        return True
    def validPalindrome(self, s: str) -> bool:
        for i in range(len(s)):
            t=s[:i]+s[i+1:]
            if(self.isPalindrome(t)) : return True
        
        return False
        